app/src/main/java/vn/mbm/phimp/me/PhimpMe.java

```diff
@@ -798,37 +796,31 @@ public void onResume() {
     }
 
     @Override
-    public boolean onKeyDown(int keycode, KeyEvent event) {
-        if (keycode == KeyEvent.KEYCODE_BACK) {
-            if (currentScreen != HomeScreenState.GALLERY) {
-                newGallery frag = new newGallery();
-                getSupportFragmentManager().beginTransaction()
-                        .replace(R.id.fragment_container, frag)
-                        .commit();
-                currentScreen = HomeScreenState.GALLERY;
-            }
-            else {
-                AlertDialog.Builder alertbox = new AlertDialog.Builder(ctx);
-                alertbox.setMessage(getString(R.string.exit_message));
-                alertbox.setPositiveButton(getString(R.string.yes), new DialogInterface.OnClickListener() {
-                    @Override
-                    public void onClick(DialogInterface dialog, int which) {
-                        finish();
-                        System.exit(0);
-                    }
-                });
-                alertbox.setNegativeButton(getString(R.string.cancel), new DialogInterface.OnClickListener() {
-                    @Override
-                    public void onClick(DialogInterface dialog, int which) {
-                        //Resume to current process
-                    }
-                });
-                alertbox.create().show();
-
-            }
+    public void onBackPressed() {
+        if (currentScreen != HomeScreenState.GALLERY) {
+            newGallery frag = new newGallery();
+            getSupportFragmentManager().beginTransaction()
+                    .replace(R.id.fragment_container, frag)
+                    .commit();
+            currentScreen = HomeScreenState.GALLERY;
+        } else {
+            AlertDialog.Builder alertbox = new AlertDialog.Builder(this);
+            alertbox.setMessage(getString(R.string.exit_message));
+            alertbox.setPositiveButton(getString(R.string.yes), new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(DialogInterface dialog, int which) {
+                    PhimpMe.this.onBackPressed();
+                }
+            });
+            alertbox.setNegativeButton(getString(R.string.cancel), new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(DialogInterface dialog, int which) {
+                    //Resume to current process
+                }
+            });
+            alertbox.create().show();
 
         }
-        return super.onKeyDown(keycode, event);
```

Comment:
The entire method has been changed. So we are using onBackPressed instead of onKeyDown